src/libmongoc/tests/test-mongoc-client-session.c

```diff
@@ -2224,8 +2222,8 @@ test_cursor_implicit_session (void *ctx)
 
    /* push a new server session into the pool */
    mongoc_client_session_destroy (cs);
```

Comment:
I think we should use the `cs` client session so it does get added to the pool. You can use it by appending it to a command's options. With that, I think we can keep these assertions the same.